app/controllers/transactions_controller.rb

```diff
@@ -0,0 +1,38 @@
+class TransactionsController < ApplicationController 
+  def index
+    @transactions = Transaction.all
+  end
+
+  def show 
+    @transaction = Transaction.find(params[:id])
+    @string_to_hash = JSON.parse(@transaction.data)
+
+    obj = CategoryAnalyzer.new(@string_to_hash)
+    @per_category = obj.expenses_per_category
+    @per_subcategory = obj.expenses_per_subcategory
```

Comment:
Move this code block in the transaction helper as it is only being used in the views.

Reply:
done